notmuch broken on latest unstable

**Describe the bug**
notmuch fails to build in the unstable branch, its test suite fails and reports an error

**To Reproduce**
`nixos-rebuild` with notmuch somewhere in your environment
**Additional context**
```
Notmuch test suite complete.
1207/1224 tests passed.
3 broken tests failed as expected.
14 tests failed.
All tests in 1 file skipped.
make: *** [test/Makefile.local:75: test] Error 1
builder for '/nix/store/lqv0vpl61i0ccbwgqi4z4y7pnm9iaxbm-notmuch-0.31.drv' failed with exit code 2
cannot build derivation '/nix/store/farp52vkpmnsjfhinrp84xzcdkkjv70s-man-paths.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/mlcsajhrc109hgxll520bdskw0z5mnvw-neomutt-20201127.drv': 1 dependencies couldn't be built
```

**Notify maintainers**
@flokli @puckipedia 

**Metadata**
```
 - system: `"x86_64-linux"`
 - host os: `Linux 5.9.16-hardened, NixOS, 21.03pre260787.56bb1b0f7a3 (Okapi)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.10`
 - channels(root): `"nixos-21.03pre262347.257cbbcd3ab, nixos-hardware"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute: notmuch
# a list of nixos modules affected by the problem
module: 
```

--- a/pkgs/applications/networking/mailreaders/notmuch/default.nix
+++ b/pkgs/applications/networking/mailreaders/notmuch/default.nix
@@ -86,6 +86,12 @@ stdenv.mkDerivation rec {
     gdb man emacs
   ];
 
+  # Expects there to always be a thread with ID
+  # thread:0000000000000009, but notmuch new is non-deterministic so
+  # this isn't always the case.  Upstream bug report:
+  # https://nmbug.notmuchmail.org/nmweb/show/871reno6g7.fsf%40alyssa.is
+  NOTMUCH_SKIP_TESTS = "lib-thread";
+
   installTargets = [ "install" "install-man" "install-info" ];
 
   postInstall = stdenv.lib.optionalString withEmacs ''

